### ISSUE TITLE
Bren carrier's middle wheels move very slowly compared to the other wheels

### ISSUE THREAD

**Issue by handsm** [DarklightGames/DarkestHour#692]
The carrier has three equally sized road wheels each side, as well as the raised up sprocket and idler wheels. The middle road wheels rotate very slowly - visibly much slower than the neighbouring road wheels. This is very odd as the wheel animation code rotates all wheels at equal speed.

**Comment by handsm** [comment]
Some background: there are two types of wheels on a tracked vehicle. First the wheels that you can see and which are rotated by code in a vehicle's Tick() function, but which are cosmetic only and have no effect on driving. Second the physics wheels, which are invisible but are actually driving the vehicle, being rotated by native Unreal engine code. Each of the individual wheels, both cosmetic and physics, have their own bone in the vehicle's hull mesh rig.

A fully tracked vehicle drives on six physics wheels, three each side, and typically about the height of the tracks. The front wheels steer, the middle ones are drive only and the rear wheels inverse steer, all creating tank-like driving. The treads are only cosmetic, like the visible wheels, and don't effect driving.

The problem is because the carrier is set up to have six physics wheels, but only has four bones for them. It doesn't have bones for the middle drive-only wheels, probably because the tracks are so short compared to other tracked vehicles and the modeller/rigger felt that four wheels was enough. But when the vehicle's default properties were set up, six physics wheels have been specified and the middle ones  'borrow' the bones for the cosmetic middle road wheels, so those bones get used twice. This set up goes back to the RO Universal Carrier.

Unfortunately the native physics wheel code rotates its specified bone, which is used to actually move the vehicle. It needs its own, dedicated bone. By using the same bone for a physics wheel and a cosmetic wheel it gets rotated twice; once by native physics code and again by script code, when the cosmetic wheel is 'animated'. This certainly screws up the rotation speed of the cosmetic middle road wheels. And I'm not sure whether the script adjustment to the rotation of a bone used by a physics wheel has an effect on driving, but it's certainly a bad idea to mess around with it.

**Comment by handsm** [comment]
Fixed in commit https://github.com/DarklightGames/DarkestHour/commit/bb3273f86d9da46d5b54d21b2f59f8dc4c0fa8ae.

Haven't altered the rig of the carrier as that would require new driver animations. Have simply deleted the middle drive-only physics wheels from the vehicle's default properties, so it runs on four physics wheels, probably how the modeller/rigger intended it. The tracks are very short and four wheels seems sufficient.

Repositioned the physics wheels on the front and rear tread contact points, i.e. the furthest points where the treads touch the ground. Changed the physics wheel diameter to be approximately the height of the tracks, which is how I set up tracked vehicles and they seem to drive pretty smoothly. Set the height position of the physics wheels so the treads rest level on the ground. Adjusted the physics centre of mass so it is just above the ground, which seems to give best stability to vehicles, and the carrier is balanced evenly front and back. Adjusted transmission ratio to give same speed as before (due to physics wheel size changing).

Enabled the bSpecialTankTurning property, the same as for tanks and all other fully tracked vehicles. This makes sense as the carrier is fully tracked and will steer differently to a half-tracked vehicle. The RO carrier had bSpecialTankTurning, but it was changed in DH. Imo the carrier feels more realistic when it turns now. Previously it spun very quickly and to me felt all wrong for a tracked vehicle.
